### PR TITLE
[fix](scanner scheduler) fix coredump of ScannerScheduler::_scanner_scan

### LINKED ISSUE #15198: [Bug]  scanner_scan SIGSEGV address not mapped to object (@0x0)
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master fdb54a346d854b6665cb2631639ba8f04b5dee14

### What's Wrong?

The following sql cause BE coredump:
```
select
          name
      from
          tpch_tiny_nation
      where
          split_part("bCKHDX07at", "5.7.37", cast(name as int)) is not null;
```

coredump stack:
```
*** Query id: dab461c704f14245-9da008284747d089 ***
*** Aborted at 1671467872 (unix time) try "date -d @1671467872" if you are using GNU date ***
*** Current BE git commitID: 494eb895d ***
*** SIGSEGV address not mapped to object (@0x0) received by PID 975791 (TID 0x7f2bbeafa700) from PID 0; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /home/zcp/repo_center/doris_master/doris/be/src/common/signal_handler.h:420
 1# 0x00007F2C89D29BF9 in /usr/lib/jvm/java-1.11.0-openjdk-amd64/lib/server/libjvm.so
 2# JVM_handle_linux_signal in /usr/lib/jvm/java-1.11.0-openjdk-amd64/lib/server/libjvm.so
 3# 0x00007F2C89D2293C in /usr/lib/jvm/java-1.11.0-openjdk-amd64/lib/server/libjvm.so
 4# 0x00007F2C8FCE60C0 in /lib/x86_64-linux-gnu/libc.so.6
 5# doris::vectorized::Block::bytes() const at /home/zcp/repo_center/doris_master/doris/be/src/vec/core/block.cpp:362
 6# doris::vectorized::ScannerScheduler::_scanner_scan(doris::vectorized::ScannerScheduler*, doris::vectorized::ScannerContext*, doris::vectorized::VScanner*) at /home/zcp/repo_center/doris_master/doris/be/src/vec/exec/scan/scanner_scheduler.cpp:252
 7# doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}::operator()() const at /home/zcp/repo_center/doris_master/doris/be/src/vec/exec/scan/scanner_scheduler.cpp:144
 8# void std::__invoke_impl<void, doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}&>(std::__invoke_other, doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}&) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:61
 9# std::enable_if<is_invocable_r_v<void, doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}&>, void>::type std::__invoke_r<void, doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}&>(doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}&) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:117
10# std::_Function_handler<void (), doris::vectorized::ScannerScheduler::_schedule_scanners(doris::vectorized::ScannerContext*)::{lambda()#1}>::_M_invoke(std::_Any_data const&) at /var/local/ldb_toolchain/include/c++/11/bits/std_function.h:292
11# std::function<void ()>::operator()() const at /var/local/ldb_toolchain/include/c++/11/bits/std_function.h:560
12# doris::FunctionRunnable::run() at /home/zcp/repo_center/doris_master/doris/be/src/util/threadpool.cpp:46
13# doris::ThreadPool::dispatch_thread() at /home/zcp/repo_center/doris_master/doris/be/src/util/threadpool.cpp:543
14# void std::__invoke_impl<void, void (doris::ThreadPool::*&)(), doris::ThreadPool*&>(std::__invoke_memfun_deref, void (doris::ThreadPool::*&)(), doris::ThreadPool*&) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:74
15# std::__invoke_result<void (doris::ThreadPool::*&)(), doris::ThreadPool*&>::type std::__invoke<void (doris::ThreadPool::*&)(), doris::ThreadPool*&>(void (doris::ThreadPool::*&)(), doris::ThreadPool*&) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:97
16# void std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>::__call<void, , 0ul>(std::tuple<>&&, std::_Index_tuple<0ul>) at /var/local/ldb_toolchain/include/c++/11/functional:422
17# void std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>::operator()<, void>() at /var/local/ldb_toolchain/include/c++/11/functional:505
18# void std::__invoke_impl<void, std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>&>(std::__invoke_other, std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>&) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:61
19# std::enable_if<is_invocable_r_v<void, std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>&>, void>::type std::__invoke_r<void, std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>&>(std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()>&) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:117
20# std::_Function_handler<void (), std::_Bind<void (doris::ThreadPool::*(doris::ThreadPool*))()> >::_M_invoke(std::_Any_data const&) at /var/local/ldb_toolchain/include/c++/11/bits/std_function.h:292
21# std::function<void ()>::operator()() const at /var/local/ldb_toolchain/include/c++/11/bits/std_function.h:560
22# doris::Thread::supervise_thread(void*) at /home/zcp/repo_center/doris_master/doris/be/src/util/thread.cpp:455
23# start_thread at /build/glibc-sMfBJT/glibc-2.31/nptl/pthread_create.c:478
24# __clone in /lib/x86_64-linux-gnu/libc.so.6
```

### What You Expected?

no coredump.

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jacktengg**
# Proposed changes

Issue Number: close #15198

## Problem summary

Introduced by #14997
`ScannerScheduler::_scanner_scan` missed checking returned Status of `scanner->get_block`, when `scanner->get_block` failed some column is NULL, which caused coredump when calling `block->bytes()`.

The sql mentioned in issue returns `Status::RuntimeError` in `FunctionSubstringIndex:: execute_impl`:
```
class FunctionSubstringIndex : public IFunction {
   Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
                            size_t result, size_t input_rows_count) override {

        for (size_t i = 1; i <= 2; i++) {
            ColumnPtr columnPtr = remove_nullable(block.get_by_position(arguments[i]).column);

            if (!is_column_const(*columnPtr)) {
                return Status::RuntimeError("Argument at index {} for function {} must be constant",
                                            i + 1, get_name());
            }
        }
    }
}
```

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.18 seconds
 load time: 628 seconds
 storage size: 17123649045 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221220074500_clickbench_pr_65506.html

**Review by Gabriel39 — DISMISSED** [review]
LGTM

File: be/src/vec/exec/scan/scanner_scheduler.cpp [hunk 1]
```diff
@@ -249,6 +249,13 @@ void ScannerScheduler::_scanner_scan(ScannerScheduler* scheduler, ScannerContext
             eos = true;
         }
 
+        if (UNLIKELY(!status.ok())) {
```
Comment (morningman) [inline be/src/vec/exec/scan/scanner_scheduler.cpp:252]: I think this logic should be merged with `if` in line 238?

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**